src/math/decibels.c

```diff
@@ -105,10 +72,10 @@ int32_t exp_fixed(int32_t x)
 		n++;
 	}
 
-	/* exp_small_fixed() input is Q3.29, while x1 is Q5.27
-	 * exp_small_fixed() output is Q9.23, while y0 is Q12.20
+	/* sofm_exp_int32() input is Q4.28, while x1 is Q5.27
+	 * sofm_exp_int32() output is Q9.23, while y0 is Q12.20
 	 */
-	y0 = Q_SHIFT_RND(exp_small_fixed(Q_SHIFT_LEFT(xs, 27, 29)), 23, 20);
+	y0 = Q_SHIFT_RND(sofm_exp_int32(Q_SHIFT_LEFT(xs, 27, 28)), 23, 20);
```

Comment:
line 78 - 81 still can be optimized further with instrinsic code.

Reply:
this file didn't include xtensa header file, do you mean add hifi implementation of functions in this file?

Reply:
yes, each for loop deserve an intrinsic implementation.

Reply:
Please check if it improves speed. In some cases the C multiply has been as fast.

Reply:
I'll give { `Q_CONVERT_FLOAT` ~, { `Q_SHIFT_RND` ~, { `Q_SHIFT_LEFT` ~, and `Q_MULTSR_32X32` } a shot as well. If I don't succeed, I'll try again with the next PR.

Reply:
![image](https://github.com/thesofproject/sof/assets/61411947/9cde8849-add0-4f3c-813d-a9eba6fabf86)

Reply:
I have completed the implementation of Q_SHIFT_RND, Q_SHIFT_LEFT, and Q_MULTSR_32X32, and I am currently working on Q_CONVERT_FLOAT.

Reply:
I'll correct the last few CI errors.

Reply:
I've finished addressing every CI error.

Reply:
@ShriramShastry , still code style error.